lib/spec/spec.go

```diff
@@ -99,8 +102,17 @@ func NewVirtualMachineConfigSpec(ctx context.Context, session *session.Session,
 	log.Debugf("Adding metadata to the configspec: %+v", config.Metadata)
 	// TEMPORARY
 
+	// set VM name to prettyname-ID, to make it readable a little bit
+	// if prettyname-ID is longer than max vm name length, truncate pretty name, instead of UUID, to make it unique
+	nameMaxLen := MaxVMNameLength - len(config.ID)
+	prettyName := config.Name
+	if len(prettyName) > nameMaxLen-1 {
+		prettyName = prettyName[:nameMaxLen-1]
+	}
+	fullName := fmt.Sprintf("%s-%s", prettyName, config.ID)
 	s := &types.VirtualMachineConfigSpec{
-		Name: config.ID,
+		Name:         fullName,
+		InstanceUuid: config.InstanceUUID,
```

Comment:
Yes, I meant Uuid.  See: http://pubs.vmware.com/vsphere-60/index.jsp#com.vmware.wssdk.apiref.doc/vim.SearchIndex.html#findByUuid

> If specified, search for virtual machines whose instance UUID matches the given uuid. 
> Otherwise, search for virtual machines whose BIOS UUID matches the given uuid.